### ISSUE TITLE
pestpp-ies error writing model input files from template files

### ISSUE THREAD

**Issue by dpphat** [usgs/pestpp#18]
Hi all.
I am having an issue in which I am getting the following error:

**model input/output error: error writing model input files from template files internal error condition has arisen while attempting to write current value of parameter to model input file**

If I run pestpp-ies with NOPTMAX = 0, I do not get this error and the model runs as expected. When I set NOPTMAX to 3 (or any other positive integer) I get this error. The error causes the model to abort and it retries.

Any guidance related to how to debug/resolve the issue is much appreciated.
Thanks so much.
Dan
[pestpp-ies error writing model input files from template files.zip](https://github.com/usgs/pestpp/files/3423992/pestpp-ies.error.writing.model.input.files.from.template.files.zip)

**Comment by jtwhite79** [comment]
@dpphat - I'm looking into this issue now

**Comment by dpphat** [comment]
Thanks so much!

**Comment by jwhite-usgs** [comment]
@dpphat - can you post the control file also?

**Comment by dpphat** [comment]
For sure.
It is attached.
Thanks again for looking into this.
[calibration.pst.txt](https://github.com/usgs/pestpp/files/3428975/calibration.pst.txt)

**Comment by jwhite-usgs** [comment]
sorry @dpphat - one more request:  can you post prior.jcb also? It must be something with the values in that initial par ensemble...

**Comment by dpphat** [comment]
No sorry at all. I really appreciate you looking into this.
The prior.jcb is attached.
[prior.jcb.txt](https://github.com/usgs/pestpp/files/3429601/prior.jcb.txt)

**Comment by jwhite-usgs** [comment]
alright @dpphat - found the issue.  That prior.jcb has some negative values for log-transformed pars, which then produces denormal numbers when the log transform is applied.  Admittedly, this should be caught during the ensemble read instead at model run time.  Im adding checks for denormal numbers in the initialization routine and during run mgmt now.

How did you make this prior.jcb?

**Comment by dpphat** [comment]
Dang. Ok.
I made the prior .jcb using ens.to_binary from pyEmu, where ens is my ensemble. 

I did not use ens.enforce() because I found that that function replaced all the parameter values with two values. That is to say that each histogram for parameter values produced two bars at opposite ends of the scale. When I did not use ens.enforce() I got a nice distribution of parameter values. Unfortunately, as you are pointing out, the result is that parameters are now out of bounds.

I will look into why I am getting these weird results from ens.enforce. At least if I am not able to figure it out I can at least re-sample. Thank you so much for your help with this.
